source/directives/uiTreeNode.js

```diff
@@ -63,7 +63,7 @@
             });
 
             scope.$watch(attrs.expandOnHover, function(val) {
-              if ((typeof val) == 'boolean') {
+              if ((typeof val) == 'boolean' || (typeof val) == 'number') {
```

Comment:
@StefanCodes was this [the regression issue you've mentioned](https://github.com/angular-ui-tree/angular-ui-tree/pull/806#issuecomment-244822788)?